app/services/base_service.rb

```diff
@@ -93,6 +93,20 @@ def initialize(result, message:)
     end
   end
 
+  class PaymentProviderFailure < FailedResult
```

Comment:
Since we are relying on many integration, I would suggest to rename the `ThirdPartyFailure` into `IntegrationFailure` and allow it to expose the initial error from the provider.

In many places in the project we have different ways of dealing with provider error, I'm not sure that adding a new one will simplify it :smile: